From 5e51b25da971e8820ee8010d6ce8666bc3f5bc04 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Sun, 5 May 2019 16:22:33 +0800
Subject: [PATCH 1/8] net:fec:reseve pps interrupt for fec_ptp driver on imx7d

imx7d defined 4 interrupts in device-tree, the last is for pps.
This interrupt is reserved by fec_ptp driver. fec ethernet driver
can not use this interrupt.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/fec_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index 17660aa6da95..c2546f2b602c 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -3606,6 +3606,8 @@ static int fec_enet_get_irq_cnt(struct platform_device *pdev)
 
 	if (irq_cnt > FEC_IRQ_NUM)
 		irq_cnt = FEC_IRQ_NUM;	/* last for pps */
+	else if (irq_cnt == 4)
+		irq_cnt = 3; /*last for pps on imx7*/
 	else if (irq_cnt == 2)
 		irq_cnt = 1;	/* last for pps */
 	else if (irq_cnt <= 0)
-- 
2.17.1

